packages/core/src/utils.js

```diff
@@ -220,10 +220,22 @@ function computeDefaults(
     );
   } else if ("oneOf" in schema) {
     schema =
-      schema.oneOf[getMatchingOption(undefined, schema.oneOf, rootSchema)];
+      schema.oneOf[
+        getMatchingOption(
+          JSON.stringify(formData) !== "{}" ? formData : undefined,
+          schema.oneOf,
+          rootSchema
+        )
+      ];
   } else if ("anyOf" in schema) {
     schema =
-      schema.anyOf[getMatchingOption(undefined, schema.anyOf, rootSchema)];
+      schema.anyOf[
+        getMatchingOption(
+          JSON.stringify(formData) !== "{}" ? formData : undefined,
```

Comment:
Also, should we move this logic out of this file and directly inside the `getMatchingOption` function instead?

Finally, `getMatchingOption` is also [used a few times in MultiSchemaField.js](https://github.com/rjsf-team/react-jsonschema-form/blob/3ec17f1c0ff40401b7a99c5e9891ac2834a1e73f/packages/core/src/components/fields/MultiSchemaField.js#L21-L33) -- should we apply this fix there as well, and if we do, perhaps we could add a test that covers that change as well?

Reply:
I'm a bit scared to widen this change.

I know that `getMatchingOption` is used for a couple of different purposes, such as generating defaults (as we're doing here), and for loading the initial selected multi-schema options (as you mentioned in `MultiSchemaField`), we also use it in `resolveDependencies`.

These use cases have subtle differences - in the `computeDefault` function, `formData` gets defaulted to `{}` (for reasons I don't understand), and that's why we need this check here so it goes back to generating defaults. Whereas in other places it may be necessary to keep the empty object as the object type may be matched to a particular sub-schema.

Reply:
If I may, I will restrain myself from potentially breaking the world. I'll leave it to your judgement as you are the expert on this.

Reply:
@epicfaace is this a blocking comment? It would be great to get this pr merged.

Reply:
cc @nickgros (I see you on an active pr)